src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -833,34 +833,22 @@ async def disk_choices(self):
         """
         diskchoices = {}
 
-        zvol_query_filters = [('type', '=', 'DISK')]
-
-        used_zvols = [
-            i['path'] for i in (await self.query(zvol_query_filters))
-        ]
-
         zvols = await self.middleware.call(
-            'pool.dataset.query', [
-                ('type', '=', 'VOLUME'),
-                ('locked', '=', False)
```

Comment:
How does this new approach handle locked datasets?

Reply:
Locked datasets aren't available through /dev/zvol (handled by ZFS).